### PR TITLE
refactor: ensures dependencies are up-to-date

### PR DESCRIPTION
**Pull request by BWsix**
Fixes https://github.com/line/create-liff-app/issues/11

This PR changes how `create-liff-app` install dependencies, ensuring that all dependencies used in generated projects are up-to-date. I'm not sure if this is fixing the issue or making it even worse, any feedback would be great🙏

Changes:
- `.eslinerc.js`
  - removed `'linebreak-style': ['error', 'unix']` since all the linebreaks are in `CRLF`
- removed `dependencies` and `devDependencies` fields in every `package.json` in `/templates` (**except for nuxtjs templates**)
- removed `@line/create-liff-app` upgradation script
- `create-liff-app.ts`
  - added `templates` object in the end of the file to store all the dependencies used by different frameworks
  - moved `envPrefix` and `envFileNameVariants` into `templates` object
  - `install` function now installs dependencies from `templates` object in runtime

Problems that need to be discussed:
- (not related to this PR) nuxtjs projects generated by the current version(1.0.4) of `create-liff-app` do not work as expected, as it always throws this error: `Error: liffId is necessary for liff.init()`. I do not know how to fix that.
- under the current refactor, when a user selects `no` on the question `install now`, `create-liff-app` will first run `npm install --package-lock-only` to install the latest version of the packages without actually saving them to the node_modules, then use `fs.rmSync` to delete the generated `package-lock.json`.
  I do not think this is a good way of handling the problem and maybe it needs to be replaced? Below are two possible fixes that I came up with. What do you think?
  - remove the `install now` option and install the dependencies by default list `create-next-app` does. This is lesserror-prone.
  - bring back `dependencies` and `devDependencies` fields in every `package.json` in `/templates`. Use them to generate outdated dependencies as it does in the current version(1.0.4) of `create-liff-app`, then remind the user to upgrade the dependencies in the `showDoneComments` function.

Notes:
- `Nuxtjs`
  - dependencies used in `create-nuxt-app` are hardcoded into `package.json`, so this refactor followed the conversion and only `@line/liff` is installed in the runtime

### CODE REVIEW

**Comment by BWsix** [comment]
I just realized that if the user decided not to install the packages right away, there will be no dependencies in the `package.json` generated.
I'll close this PR and work on that right now.
Probably going to use `--package-lock-only`

File: create-liff-app.ts [hunk 1]
```diff
@@ -275,3 +300,50 @@ const questions: Array<Question | ListQuestion> = [
...
 ];
+
+type TemplateOptions = {
+  envPrefix: string;
+  envFileNameVariant?: string;
+  dependencies: string[];
+  devDependencies: string[];
+  tsDevDependencies: string[];
+};
+const templates: Record<string, TemplateOptions> = {
```
Comment (so99ynoodles) [inline create-liff-app.ts:311]: 👍

File: create-liff-app.ts [hunk 2]
```diff
@@ -96,14 +83,25 @@ export async function createLiffApp(answers: Answers) {
...
     fs.writeFileSync(path.join(root, envFileName), content);
 
     // install
     const isYarn = packageManager === 'yarn';
-    if (installNow) {
-      await install(root, isYarn);
+    const { dependencies, devDependencies, tsDevDependencies } = templateConfig;
+    if (isTypescript) devDependencies.push(...tsDevDependencies);
+
+    console.log(`\n${installNow ? 'Installing' : 'Updating'} dependencies:`);
```
Comment (so99ynoodles) [inline create-liff-app.ts:95]: > remove the install now option and install the dependencies by default list create-next-app does. This is lesserror-prone.

As you mentioned, we can delete `installNow` I think.

**Review by so99ynoodles — COMMENTED** [review]
@BWsix 
Sorry for the late review 🙇 
Looks good overall 🎉 . Thank you for the great work.

I'm a little concerned about `Nuxtjs ` workaround.
Since `Nuxtjs` and `Nextjs` provide their own `create-app` feature, how do you think using them inside `create-liff-app` and replacing the result with our templates?
@LeeChSien @cola119 @sugarshin

File: create-liff-app.ts (hunk 2 above)
Comment (BWsix) [reply create-liff-app.ts:95]: ok, I'll remove this feature.

**Review by sugarshin — COMMENTED** [review]
Overall looks good to me. However, let us set up CI to automatically test and lint before merging.

> Since Nuxtjs and Nextjs provide their own create-app feature, how do you think using them inside create-liff-app and replacing the result with our templates?

@so99ynoodles It looks good to me.

File: create-liff-app.ts [hunk 3]
```diff
@@ -47,40 +29,25 @@ export function init() {
...
-        key: 'npm',
-        value: 'npm',
-        checked: false,
-      },
-    ],
-  }).then(({ packageManager }) => packageManager as PackageManager) : 'npm';
-  
-  const templateName = `${template}${language === 'JavaScript' ? '' : '-ts'}`;
+  const { projectName, template, language, liffId } = answers;
+  const templateConfig = templates[template];
```
Comment (cola119) [inline create-liff-app.ts:39]: Sorry for nitpicking but let's make more type-safe

```suggestion
  const templateConfig = templates[template] as TemplateOptions | undefined;
  if (!templateConfig) {
    throw new Error(`Invalid template name: ${template}`);
  }
```

File: create-liff-app.ts [hunk 4]
```diff
@@ -204,7 +194,7 @@ const questions: Array<Question | ListQuestion> = [
         console.log(`\n${chalk.yellow('The project is already exists.')}`);
         return false;
       }
-      
+
```
Comment (cola119) [inline create-liff-app.ts:197]: Could you remove this unnecessary change?

**Review by cola119 — COMMENTED** [review]
I left some nits comments but overall LGTM, great work!

File: create-liff-app.ts (hunk 3 above)
Comment (BWsix) [reply create-liff-app.ts]: of course!

File: create-liff-app.ts (hunk 4 above)
Comment (BWsix) [reply create-liff-app.ts:200]: ok, I think that was removed by the formatter.

**Comment by BWsix** [comment]
> Since Nuxtjs and Nextjs provide their own create-app feature, how do you think using them inside create-liff-app and replacing the result with our templates?

I did some changes and create-liff-app will now generate nextjs & nuxtjs projects using their `create-app` feature. Does this look ok to you?

Changes:
- after users answer the questions, create-liff-app will now generate nextjs / nuxtjs project using `create-app`. Then, create-liff-app will use its template to overwrite the generated files.
- removed most of the files in `/templates/nextjs` & `/templates/nuxtjs` since they are the same as the `create-app` generated ones.

> nuxtjs projects generated by the current version(1.0.4) of create-liff-app do not work as expected, as it always throws this error: Error: liffId is necessary for liff.init(). I do not know how to fix that.

Also, as I mentioned in the original PR, nuxtjs app does not work as expected. Therefore, I am unsure if the above change will break anything. Please remember to double-check the changes!

**Comment by BWsix** [comment]
> let us set up CI to automatically test and lint before merging

Please help me out on this one, as I do not know how😅

**Review by so99ynoodles — APPROVED** [review]
@BWsix 
> I did some changes (https://github.com/line/create-liff-app/commit/c406a40b82db1ffac6a30553329bc0f6d239ce8f) and create-liff-app will now generate nextjs & nuxtjs projects using their create-app feature. Does this look ok to you?

This change looks pretty good! 👍 LGTM
Let us talk again within the team and release this PR after that.
Thank you for your awesome work.

> let us set up CI to automatically test and lint before merging

I think we can separate it as another issue.
#16

**Comment by BWsix** [comment]
Thank you and your team for the time and effort you spent reviewing this PR👍

**Review by sugarshin — APPROVED** [review]
LGTM

> Overall looks good to me. However, let us set up CI to automatically test and lint before merging.

Let us handle it on another PR. Thank you for your contribution.

**Comment by so99ynoodles** [comment]
Maybe we should fix the test since the local test fails. `yarn test`

**Comment by so99ynoodles** [comment]
Merging this! 🎉
